qiskit/transpiler/passes/routing/star_prerouting.py

```diff
@@ -11,34 +11,31 @@
 # that they have been altered from the originals.
 
 """Search for star connectivity patterns and replace them with."""
-from typing import Iterable, Union, Optional, List, Tuple
+from typing import Union, List, Tuple
 from math import floor, log10
 
 from qiskit.circuit import Barrier
 from qiskit.dagcircuit import DAGOpNode, DAGDepNode, DAGDependency, DAGCircuit
+from qiskit.dagcircuit.collect_blocks import Block, BlockCollector
 from qiskit.transpiler import Layout
 from qiskit.transpiler.basepasses import TransformationPass
 from qiskit.circuit.library import SwapGate
 
 
-class StarBlock:
+class StarBlock(Block):
```

Comment:
This and other comments regarding `star_preroute.py` are no longer relevant after in [1a5ef91](https://github.com/Qiskit/qiskit/pull/12498/commits/1a5ef91b27d920b8c582bac2cab10805a81aec19) I've reverted the changes to this file, in order not to interfere with #12761.

I completely agree that it would be best to create a "library" of useful and reusable block types (like `DefaultBlock`). I am not completely convinced that specifically `StarBlock` is going to be useful beyond the `StarPrerouting` pass, so I am not sure we should expose this particular block type. In addition, I have locally tried to improve the quality of `StarPreroute` pass by experimenting with different variants of the `StarBlock` (for example only collecting stars with non-repeating non-center nodes, the intuition being that repeating nodes prevent mapping a star to a line). Unfortunately, I did not find anything that made `StarPrerouting` better on say QAOA circuits, however I do find likely that a good star block for the star prerouting pass would be very specific to this pass and thus should be packaged together with the pass.